pkg/skaffold/build/cache/cache.go

```diff
@@ -79,7 +80,7 @@ func NewCache(ctx context.Context, builder build.Builder, opts *skafconfig.Skaff
 	}
 	var imageList []types.ImageSummary
 	if client != nil {
-		imageList, err = client.ImageList(ctx, types.ImageListOptions{})
+		imageList, err = client.ImageList(context.Background(), types.ImageListOptions{})
```

Comment:
side note: https://github.com/golang/go/blob/go1.11.5/src/context/context.go#L203
```
// Background returns a non-nil, empty Context. It is never canceled, has no
// values, and has no deadline. It is typically used by the main function,
// initialization, and tests, and as the top-level Context for incoming
// requests.
func Background() Context {
 ...
}
```
`context.Backgound` most likely should be used in Main or initialization code. Hence, i was a little skeptical when i saw this here. You can say, this is initialization here, but IIUC, this is called by runner, which itself is running in background.
https://github.com/GoogleContainerTools/skaffold/blob/master/cmd/skaffold/app/cmd/build.go#L69
```
func runBuild(out io.Writer) error {
	start := time.Now()
        ... 

	ctx, cancel := context.WithCancel(context.Background())
        ... 
        bRes, err := runner.BuildAndTest(ctx, buildOut, targetArtifacts)
```
I feel we should go back to previous sig where context is passed in. That way it also gives the flexibility later when some other command wants to build cache in fore-ground.
Does that make sense?

Reply:
It does make sense &  I did prefer it the original way as well, but I need to access the skaffold config to determine if images are being built locally, which would require either:

1. making the change I made in this PR and passing in context.Background as a placeholder
2. adding the skaffold config to the runner struct so it can be passed in during build, which felt a bit weirder (but am happy to do that if everyone else feels it's better)

I figured option 1 was better because this is only needed to initialize the cache; later, when cache functions are called, `ctx` will be correctly passed in. 

WDYT?

Reply:
yeah i think this is another instance where we need to create a  `RunnerContext` and then pass it to build and elsewhere as discussed @balopat.
I think its fine to keep this as is since we decided to work on something like this even for  #1870